### ISSUE TITLE
Uncaught TypeError: arr.where is not a function

### ISSUE THREAD

**Issue by jrwren** [joaom182/linqjs#2]
chrome Version 46.0.2489.0 canary (64-bit)

 Uncaught TypeError: arr.where is not a function(anonymous function) @ linq.js:260Array.innerJoin @ linq.js:259(anonymous function) @ join.js:40(anonymous function) @ join.js:42

given this use:

    const g1 = [{ "Name": "Alice", "Hair": "Blonde"},
              { "Name": "Bob", "Hair": "None"},
    ];
    const g2 = [{ "Name": "Alice", "Hair": "Brown"},
              { "Name": "Bob", "Hair": "Brown"},
    ];
    const tada = g1.innerJoin( e => e.Name, e => e.Name, (e1, e2) => ({ "Name": e1.Name, "Hair1": e1.Hair, "Hair2": e2.Hair }));

**Comment by aluanhaddad** [comment]
There is a bug in your example.
```JavaScript
const tada = g1.innerJoin( e => e.Name, e => e.Name, (e1, e2) => ({ "Name": e1.Name, "Hair1": e1.Hair, "Hair2": e2.Hair }));
```
needs to be
```JavaScript
const tada = g1.innerJoin(g2, e => e.Name, e => e.Name, (e1, e2) => ({ "Name": e1.Name, "Hair1": e1.Hair, "Hair2": e2.Hair }));
```

There is no bug in the library although the error should probably be clearer.

**Comment by aluanhaddad** [comment]
@jrwren added validation as part of #3

**Comment by joaom182** [comment]
Closed with commit 72c5d6c
